packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -1283,12 +1283,16 @@ class _InkResponseState extends State<_InkResponseStateWidget>
     assert(widget.debugCheckContext(context));
     super.build(context); // See AutomaticKeepAliveClientMixin.
 
-    Color getHighlightColorForType(_HighlightType type) {
-      const Set<MaterialState> pressed = <MaterialState>{MaterialState.pressed};
-      const Set<MaterialState> focused = <MaterialState>{MaterialState.focused};
-      const Set<MaterialState> hovered = <MaterialState>{MaterialState.hovered};
+    final ThemeData theme = Theme.of(context);
+    const Set<MaterialState> highlightableStates = <MaterialState>{MaterialState.focused, MaterialState.hovered, MaterialState.pressed};
+    final Set<MaterialState> nonHighlightableStates = statesController.value.difference(highlightableStates);
+    // Each highlightable states will be resolved separatly to get the corresponding color.
+    // For this resolution to be correct, The non-highlightable states should be preserved.
+    final Set<MaterialState> pressed = <MaterialState>{...nonHighlightableStates, MaterialState.pressed};
```

Comment:
This is kind of tricky, I hope we don't make this same kind of mistake elsewhere. I guess it's probably rare to speculatively resolve a color like this, though.